File: 14.set&map/11478.js
const fs = require("fs"); // 제출시 삭제
const path = __dirname + "/11478.txt"; // 제출시 삭제

const readline = require("readline");
const rl = readline.createInterface({
  // input: process.stdin, // 제출시 활성화
  input: fs.createReadStream(path), // 제출시 삭제
  output: process.stdout,
});

let input;
let list = [];
rl.on("line", function (line) {
  input = line;
}).on("close", function () {
  S = input;
  solution(S);
  process.exit();
});
function solution(S) {
  let answer = new Set();
  for (let i = 0; i < S.length; i++) {
    for (let j = 0; j < S.length - i; j++) {
      answer.add(S.slice(j, j + i + 1));
    }
  }
  console.log(answer.size);
}

File: 14.set&map/11478.py
from sys import stdin as s
import os

dir = os.path.dirname(os.path.realpath(__file__))+"/11478.txt"
s=open(dir,"rt")

str = s.readline().strip()
answer = set()

for i in range(len(str)):
    for j in range(len(str)):
        answer.add(str[i:i+j+1])

print(len(answer))

